banks-client/src/lib.rs

```diff
@@ -0,0 +1,253 @@
+use async_trait::async_trait;
+use futures::future::join_all;
+use solana_sdk::{
+    account::Account, clock::Slot, commitment_config::CommitmentLevel,
+    fee_calculator::FeeCalculator, hash::Hash, pubkey::Pubkey, signature::Signature,
+    transaction::Transaction, transport,
+};
+use solana_transaction_status::banks_trait::{BanksRequest, BanksResponse};
+pub use solana_transaction_status::{banks_trait::BanksClient, TransactionStatus};
+use std::io::{self, Error, ErrorKind};
+use tarpc::{
+    client, context,
+    rpc::{transport::channel::UnboundedChannel, ClientMessage, Response},
+    serde_transport::tcp,
+};
+use tokio::net::ToSocketAddrs;
+use tokio_serde::formats::Bincode;
+
+#[async_trait]
+pub trait BanksClientExt {
+    async fn send_transaction(&mut self, transaction: Transaction) -> io::Result<()>;
+    async fn get_recent_blockhash(&mut self) -> io::Result<Hash>;
+    async fn get_fees(&mut self) -> io::Result<(FeeCalculator, Hash, Slot)>;
+    async fn process_transaction_with_commitment(
```

Comment:
If we replace `commitment` with `confirmations` in `send_and_confirm_transaction`, then I propose we default `process_transaction` to 0 or 1 confirmation(s).

Then we could have:
1. `send_transaction`
2. `execute_transaction` / `process_transaction` which waits for 0 or 1 confirmation(s)
3. `settle_transaction` which waits for MAX confirmations
4. [optional] `confirm_transaction` for user-specified confirmations

cc @aeyakovenko